pages/index/index_callbacks.py

```diff
@@ -1,54 +1,285 @@
-from dash import callback
+from datetime import datetime, timedelta
+import re
+import os
+import time
+import logging
+from celery.result import AsyncResult
+import dash_bootstrap_components as dbc
+import dash
+from dash import html, callback
 from dash.dependencies import Input, Output, State
 from app import augur
-import dash
-import logging
 from cache_manager.cache_manager import CacheManager as cm
 from queries.issues_query import issues_query as iq
 from queries.commits_query import commits_query as cq
 from queries.contributors_query import contributors_query as cnq
 from queries.prs_query import prs_query as prq
-import time
-from celery.result import AsyncResult
+
 
 # list of queries to be run
 QUERIES = [iq, cq, cnq, prq]
 
-# helper function for repos to get repo_ids
-def _parse_repo_choices(repo_git_set):
-    # get repo values from repo dictionary
-    repo_dict = augur.get_repo_dict()
-    repo_values = [repo_dict[x] for x in repo_git_set]
-    # values for the repo_ids and names
-    repo_ids = [row[0] for row in repo_values]
-    repo_names = [row[1] for row in repo_values]
+# page switch login an option?
+login_enabled = os.getenv("AUGUR_LOGIN_ENABLED", "False") == "True"
 
-    return repo_ids, repo_names
 
+@callback(
+    [
+        Output("nav-login-container", "children"),
+        Output("login-popover", "is_open"),
+    ],
+    Input("augur_username_dash_persistence", "data"),
+    State("login-succeeded", "data"),
+)
+def login_username_button(username, login_succeeded):
+    """Sets logged-in-status component in top left of page.
 
-# helper function for orgs to get repo_ids
-def _parse_org_choices(org_name_set):
+    If a non-null username is known then we're logged in so we provide
+    the user a button to go to Augur. Otherwise, we redirect them to login.
 
-    org_dict = augur.get_org_dict()
-    repo_dict = augur.get_repo_dict()
+    This callback also sets a login_failure popover depending on whether
+    a requested login succeeded.
 
-    # get git urls for the repos in the organization, sum operater is used to flatten the 2D list to 1D
-    org_repos = sum([org_dict[x] for x in org_name_set], [])
-    # get repo values from repo dictionary
-    repo_values = [repo_dict[x] for x in org_repos]
-    # values for the repo_ids and names
-    org_repo_ids = [row[0] for row in repo_values]
-    org_repo_names = [row[1] for row in repo_values]
+    Args:
+        username (str | None): Username of user or None
+        login_succeeded (bool): Error enabled if login failed.
 
-    return org_repo_ids, org_repo_names
+    Returns:
+        _type_: _description_
+    """
+
+    if username:
+        navlink = [
+            dbc.NavLink(
+                f"{username}",
+                href="http://chaoss.tv:5038/account/settings",
```

Comment:
Should be generalized off of environment variables

Reply:
++ gotcha